### ISSUE TITLE
Classic Wotlk better "load more" action

### ISSUE THREAD

**Issue by davidszkl** [Auctionator/Auctionator#1232]
Please include as much of the following information as possible to help me fix the bug:

## Are you on the retail or the classic version of WoW?
Classic Wotlk
## Is this a bug or a feature request?
feature request

# Feature Request:
## Is your feature request related to a problem? Please describe
When scanning for gems for example, you only get 2-3 results, what you would like though, is to have the raw gem, and all current cuts available, to see which cut will bring the biggest margin. The problem with the load more action is that it's impossible to use because it takes ages to complete ( actually never seen it complete). The behavior I would expect from load more is that it instantly loads all variants of the gem and it's cuts, because I can just do it manually, which sucks.

**Comment by plusmouse** [comment]
How do you do it manually?

Apart from loading all the pages there isn't another way to guarantee getting all the types. You should be able to view the profit/crafting cost for a particular cut in your crafting panel:
![image](https://user-images.githubusercontent.com/60280559/196023369-f37af66e-3ca7-4d31-b9e2-ee1ca623dea9.png)
![image](https://user-images.githubusercontent.com/60280559/196023389-6db6d047-bbd2-49b6-b7a8-13009141300a.png)

**Comment by davidszkl** [comment]
Hey, what I mean is that when I do an exact search, it's instant, say there are 10-15 cuts of a gem, if I know all the names, I can manually exact search those 15 cuts and ( excluding the typing time ) I will find all 15 cuts pretty fast. But when i search a gem and then load more, it takes forever. So I wonder what's the limitation that makes the load more so slow compared to spamming single searches

**Comment by plusmouse** [comment]
Auctionator sends search text to the server and asks for the first page of results (sorted by unit price) that match. More valuable items may not show up on the first page, hence them not being in the initial generic search results.

Doing an exact search for a particular item's name ensures that the first page of results (sorted by unit price) will contain that item.

If you're doing the same searches repeatedly using a shopping list so it becomes one click (after typing them in the once) to search.

For the generic search to get all the items faster Auctionator would need to be setup with all the possible item names so that it can query them behind the scenes as multiple exact searches and then compile the results.

**Comment by davidszkl** [comment]
Yeah, it seems that this addon suffers like the other addons from some sort of slow down when looking at pages 2 and beyond.
So is it possible to make a "load all similar" where it would query the server for page 1 of every item that contains the search string ?

**Comment by plusmouse** [comment]
While is is possible, with enough work. That would require keeping a list of all possible items to search and then process to guess which pages should be searched, which is not something I'm looking to do, as that's a much more complicated search than the current one.
